cartridges/app_storefront_custom/cartridge/controllers/ContentAssets.js

```diff
@@ -0,0 +1,92 @@
+"use strict";
+
+/**
+ * @namespace ContentAssets
+ */
+
+var server = require("server");
+var cache = require("*/cartridge/scripts/middleware/cache");
+var consentTracking = require("*/cartridge/scripts/middleware/consentTracking");
+var pageMetaData = require("*/cartridge/scripts/middleware/pageMetaData");
+
+/**
+ * ContentAssets-Show : This end point will render a content asset in full storefront ContentAssets
+ * @name Base/ContentAssets-Show
+ * @function
+ * @memberof ContentAssets
+ * @param {middleware} - cache.applyDefaultCache
+ * @param {middleware} - consentTracking.consent
+ * @param {querystringparameter} - cid - the id of the content asset to be displayed in a full ContentAssets
+ * @param {category} - non-sensitive
+ * @param {renders} - isml
+ * @param {serverfunction} - get
+ */
+
+server.get(
+    "Show",
+    cache.applyDefaultCache,
+    consentTracking.consent,
+    function (req, res, next) {
+        var ContentMgr = require("dw/content/ContentMgr");
+        var Logger = require("dw/system/Logger");
+        var PageMgr = require("dw/experience/PageMgr");
+        var ContentModel = require("*/cartridge/models/content");
+        var pageMetaHelper = require("*/cartridge/scripts/helpers/pageMetaHelper");
+        var page = PageMgr.getPage(req.querystring.cid);
+
+        var currentCustomer = req.currentCustomer.raw;
+        var assetBody;
+
+        if (page != null && page.isVisible()) {
+            if (!page.hasVisibilityRules()) {
+                res.cachePeriod = 168; // eslint-disable-line no-param-reassign
+                res.cachePeriodUnit = "hours"; // eslint-disable-line no-param-reassign
+            }
+
+            res.page(page.ID, {});
+        } else {
+            var apiContent = ContentMgr.getContent(req.querystring.cid);
+
+            if (apiContent) {
+                var content = new ContentModel(apiContent, "content/assetsPage");
+                pageMetaHelper.setPageMetaData(req.pageMetaData, content);
+                pageMetaHelper.setPageMetaTags(req.pageMetaData, content);
+
+                if (currentCustomer.getProfile()) {
+                    var loggedUsersAsset = ContentMgr.getContent('logged-users');
+                    var loggedUsersAssetBody = loggedUsersAsset ? loggedUsersAsset.custom.body : '';
+                    assetBody = loggedUsersAssetBody
+                        .toString()
+                        .replace("{0}", currentCustomer.getProfile().firstName);
+                } else {
+                    var guestUsersAsset = ContentMgr.getContent('guest-users-content');
+                    var guestUsersAssetBody = guestUsersAsset ? guestUsersAsset.custom.body : '';
+                    assetBody = guestUsersAssetBody
+                        .toString()
+                        .replace("{0}", 'Guest');
```

Comment:
You can extract this replace in a function, since you are doing the same thing on line 58 and line 64.

Reply:
Done